templates/layout/base.html

```diff
@@ -24,14 +24,13 @@
             window.lazySizesConfig = window.lazySizesConfig || {};
             window.lazySizesConfig.loadMode = 1;
         </script>
-        <script async src="{{cdn 'assets/dist/theme-bundle.head_async.js'}}"></script>
+        <script async src="{{cdn 'assets/dist/theme-bundle.head_async.js' resourceHint='preload'}}"></script>
         
-        {{getFontsCollection font-display='block'}}
+        {{getFontsCollection font-display='block' resourceHint='preload'}}
         
-        <link rel="preload" href="{{cdn 'assets/dist/theme-bundle.font.js'}}" as="script">
```

Comment:
@bookernath why do you think link with rel=preload is not needed anymore? this technique makes browser to load font bundle faster. what about [browsers that doesn't support ](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/103#browser_compatibility)early hints yet?

Reply:
It's basically Chrome for now. Not sure if others support HTTP header link preloads, which would be enough.

Another option is fetchpriority="high" on the async script. That would do the same thing on supported browsers. It also helps with images. You can also add it to link preloads, which are, by default, a lower priority.

I'd keep the preloads for now. It also means you can control sequencing. e.g. the way we preload main js towards at the end of the head so that other resources get first dibs. And request it for real in the footer to guarantee it runs late.

I personally load the fonts script earlier as it's required to preload the fonts, which we want to happen ASAP.

Reply:
My assumption was that this is duplicative when we are already sending Link headers. But even if it's duplicative, I have no reason to believe it's _harmful_ (browsers will likely identify that resource is already being downloaded and not start downloading it twice... I hope). Maybe this is worth a quick test.

Cloudflare will choose when to send HTTP 103 Early Hints vs just allowing the normal Link headers to go through based on their client detection and a few other factors.

Reply:
I'd assume the browser is smart enough not to make multiple requests. That is the case for other scenarios. I'd still test.

One gotcha is that the crossorigin attribute has to be the same. If it is different, it is considered a different resource to load.

I try and add crossorigin="anonymous" if it is allowed, as it lets me gather more error data. e.g. on theme-bundle.main.js

Reply:
I have mixed views on how to deal with the main bundle with regard to loading speed. It is CPU intensive and blocking (especially with sliders) which stops user interaction (FID, INP),  and delays LCP if it starts running before the page renders. So loading it quickly increases the chance of Web Vitals issues.

For one client I deliberately delayed the main bundle from running for a short period (100ms I think) and spilt up the slider initialisation code into separate delayed tasks (setTimeout per slider). That was enough to let the CPU spend time rendering and responding to user inputs in a reasonable time (<100ms).

The thing is, does it contain stuff that is required for the page to render properly, or for interaction to work? If so, it should load and run quickly. It would be nice if the non-critical work is delayed and split into mini tasks (like my slider solution). This delay typically needs to be longer than DCL as it can happen before rendering so tasks sitting on it can delay rendering.

In some cases, my slider solution did need some CSS adjustments to make sure the sliders looked correct before they were built. This is a good move anyhow as it stops layout shifts (CLS).